mod/photos.php

```diff
@@ -1023,63 +1023,13 @@ function photos_content(&$a) {
 
 	$community_page = (($a->data['user']['page-flags'] == PAGE_COMMUNITY) ? true : false);
 
-	if((local_user()) && (local_user() == $owner_uid))
-		$can_post = true;
-	else {
-		if($community_page && remote_user()) {
-			if(is_array($_SESSION['remote'])) {
-				foreach($_SESSION['remote'] as $v) {
-					if($v['uid'] == $owner_uid) {
-						$contact_id = $v['cid'];
-						break;
-					}
-				}
-			}
-			if($contact_id) {
-
-				$r = q("SELECT `uid` FROM `contact` WHERE `blocked` = 0 AND `pending` = 0 AND `id` = %d AND `uid` = %d LIMIT 1",
-					intval($contact_id),
-					intval($owner_uid)
-				);
-				if(count($r)) {
-					$can_post = true;
-					$contact = $r[0];
-					$remote_contact = true;
-					$visitor = $cid;
-				}
-			}
-		}
-	}
-
-	// perhaps they're visiting - but not a community page, so they wouldn't have write access
-
-	if(remote_user() && (! $visitor)) {
-		$contact_id = 0;
-		if(is_array($_SESSION['remote'])) {
-			foreach($_SESSION['remote'] as $v) {
-				if($v['uid'] == $owner_uid) {
-					$contact_id = $v['cid'];
-					break;
-				}
-			}
-		}
-		if($contact_id) {
-			$groups = init_groups_visitor($contact_id);
-			$r = q("SELECT * FROM `contact` WHERE `blocked` = 0 AND `pending` = 0 AND `id` = %d AND `uid` = %d LIMIT 1",
-				intval($contact_id),
-				intval($owner_uid)
-			);
-			if(count($r)) {
-				$contact = $r[0];
-				$remote_contact = true;
-			}
-		}
-	}
+	// get the access rights for photos
+	$photos_perms = photos_permissions($owner_uid, $community_page);
 
-	if(! $remote_contact) {
-		if(local_user()) {
-			$contact_id = $_SESSION['cid'];
-			$contact = $a->contact;
+	// convert keys of the $photo_persms array into variables
+	if(count($photos_perms)) {
+		foreach ($photos_perms as $key => $value) {
+			${$key} = $value;
```

Comment:
OK will, change it. But this needs a carefully review. Originally I didn't do it because I feared to break (miss) something and getting security issues.

The other thing is that I'm not really satisfied with the name $photos_perms and the function photos_permissions() because it doesn't contain pure permissions. 
It contains rather a mix of of permissions (like $can_post) and informations about the contact (like $contact)